src/main/java/org/folio/circulation/infrastructure/storage/CirculationPolicyRepository.java

```diff
@@ -88,47 +96,33 @@ protected CirculationPolicyRepository(
     String materialTypeId = item.getMaterialTypeId();
     String patronGroupId = user.getPatronGroupId();
 
+    MultiMap params = MultiMap.caseInsensitiveMultiMap();
+    params.add(ITEM_TYPE_ID_NAME, item.getMaterialTypeId());
+    params.add(LOAN_TYPE_ID_NAME, item.determineLoanTypeForItem());
+    params.add(PATRON_TYPE_ID_NAME, user.getPatronGroupId());
+    params.add(LOCATION_ID_NAME, item.getLocationId());
+
     log.info(
       "Applying circulation rules for material type: {}, patron group: {}, loan type: {}, location: {}",
       materialTypeId, patronGroupId, loanTypeId, locationId);
 
-    final CompletableFuture<Result<Response>> circulationRulesResponse =
-      circulationRulesClient.applyRules(loanTypeId, locationId, materialTypeId, patronGroupId);
-
-    return circulationRulesResponse
-      .thenCompose(r -> r.after(this::processRulesResponse));
-  }
-
-  private CompletableFuture<Result<CirculationRuleMatch>> processRulesResponse(Response response) {
-    final CompletableFuture<Result<CirculationRuleMatch>> future = new CompletableFuture<>();
-
-    if (response.getStatusCode() == 404) {
-      future.complete(failedDueToServerError("Unable to apply circulation rules"));
-    } else if (response.getStatusCode() != 200) {
-      future.complete(failed(new ForwardOnFailure(response)));
-    } else {
-      log.info("Rules response {}", response.getBody());
-
-      String policyId = fetchPolicyId(response.getJson());
-      boolean isItemTypePresent = response.getJson().getJsonObject(APPLIED_RULE_CONDITIONS)
-        .getBoolean("materialTypeMatch");
-      boolean isLoanTypePresent = response.getJson().getJsonObject(APPLIED_RULE_CONDITIONS)
-        .getBoolean("loanTypeMatch");
-      boolean isPatronGroupPresent = response.getJson().getJsonObject(APPLIED_RULE_CONDITIONS)
-        .getBoolean("patronGroupMatch");
-
-      log.info("Policy to fetch based upon rules {}", policyId);
-
-      future.complete(succeeded(new CirculationRuleMatch(
-        policyId, new AppliedRuleConditions(isItemTypePresent, isLoanTypePresent, isPatronGroupPresent))));
-    }
-
-    return future;
+    final CompletableFuture<Result<Drools>> droolsResponse = clients.getCirculationDrools();
```

Comment:
Usually, we fetch location together with item. Can we fetch location only when it is not present on item?